internal/compiler/parser/document.rs

```diff
@@ -45,6 +46,35 @@ pub fn parse_document(p: &mut impl Parser) -> bool {
                     break;
                 }
             }
+            "@" => {
```

Comment:
I think we should make sure here that we are followed with `rust-attr` 
Otherwise this is a bit fragile and 
`@foobar` will get into this when we don't really what to.  Also the loop is quite fragile.
IMHO, we should do really parse the @rust-attr(...) there and make sure we have the right amount of ballanced parentheses in there and then use the checkpoint to start the StructDeclaration node.